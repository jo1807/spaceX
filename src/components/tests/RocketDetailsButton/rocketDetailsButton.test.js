import React from "react";
import { render, cleanup } from "@testing-library/react";
import RocketDetailsButton from "../../RocketSummary";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<RocketDetailsButton />);
  expect(asFragment()).toMatchSnapshot();
});

it('has text content saying "Details"', () => {
  const { getByTestId } = render(<RocketDetailsButton />);
  expect(getByTestId("detailsBtnId").textContent).toBe("Details");
});

//check for onClick
