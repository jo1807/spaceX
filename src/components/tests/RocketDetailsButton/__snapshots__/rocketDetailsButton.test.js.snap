// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="css-1n0vo1"
  >
    <p
      class="css-fd8g0q"
    />
    <p>
      Flight Number: 
    </p>
    <p>
      Rocket Name: 
    </p>
    <p />
    <button
      class="css-77b2el"
      data-testid="detailsBtnId"
    >
      Details
    </button>
    <div
      class="css-d04zvd"
    >
      <img
        class="css-54d50x"
      />
    </div>
  </div>
</DocumentFragment>
`;
